release.py

```diff
@@ -0,0 +1,694 @@
+
+auth = github.Auth.Token(os.environ['GITHUB_TOKEN'])
+gh_client = github.Github(auth=auth)
+
+
+VERSION_REGEX = r'(\d+\.\d+\.\d+(?:\.dev\d+)?)'
+VERSION_FILES = {
+    'ops/src': Path('ops/version.py'),
+    'ops/pyproject': Path('pyproject.toml'),
+    'testing': Path('testing/pyproject.toml'),
+    'tracing': Path('tracing/pyproject.toml'),
+    'uvlock': Path('uv.lock'),
+}
+
+
+def get_latest_release_tag(repo: github.Repository.Repository, branch_name: str) -> str | None:
+    """Get the latest release tag from the repository."""
+    releases = repo.get_releases()
+
+    maintenance_branch_version = (
+        branch_name.removesuffix('-maintenance') if branch_name.endswith('-maintenance') else None
+    )
+
+    for release in releases:
+        if release.draft:
+            continue
+
+        if maintenance_branch_version and maintenance_branch_version not in release.tag_name:
+            continue
+
+        return release.tag_name
+
+    return None
+
+
+def bump_minor_version(version: str) -> str:
+    """Bump minor version."""
+    major, minor, patch = map(int, version.split('.'))
+    return f'{major}.{minor + 1}.{patch}'
+
+
+def get_new_tag_for_release(
+    owner: str, repo: github.Repository.Repository, branch_name: str
+) -> str:
+    """Get a new tag for release.
+
+    Default value is generated by increasing the minor version of the latest release's tag.
+
+    Can be overridden with a user-provided version.
+    """
+    latest_tag = get_latest_release_tag(repo, branch_name)
+
+    suggested_tag = ''
+    if not latest_tag:
+        logger.info(f'No version tag found in branch "{branch_name}"')
+    else:
+        logger.info(f'Latest tag in branch "{branch_name}": {latest_tag}')
+        if not re.match(r'^\d+\.\d+\.\d+$', latest_tag):
+            logger.info('Latest tag is not in format X.Y.Z.')
+        else:
+            suggested_tag = bump_minor_version(latest_tag)
+            logger.info(f'Suggested new version: {suggested_tag}')
+
+    tag_prompt = f' (press enter to use the tag {suggested_tag})' if suggested_tag else ''
+    prompt = f"Input the new tag for the release{tag_prompt}, or 'c' to cancel:\n"
+    while True:
+        user_input = input(prompt).strip()
+        if user_input.lower() == 'c':
+            logger.warning('Release canceled.')
+            exit(0)
+
+        new_tag = user_input if user_input else suggested_tag
+
+        if not new_tag:
+            logger.error('Error: No tag specified and no suggestion available')
+            continue
+
+        if not re.match(r'^\d+\.\d+\.\d+$', new_tag):
+            logger.error('Error: Tag must be in format X.Y.Z')
+            continue
+
+        release_page = f'https://github.com/{owner}/{repo.name}/releases'
+        logger.warning(f'Check out the releases page: {release_page} before confirming!')
+
+        confirm = (
+            input(f"Confirm creating tag '{new_tag}' on branch '{branch_name}'? [y/N]: ")
```

Comment:
You can get rid of all the extra quotes and have Python add them for you by using the `!r` format (repr). Similar elsewhere:

```suggestion
            input(f'Confirm creating tag {new_tag!r} on branch {branch_name!r}? [y/N]: ')
```